src/edu/csus/ecs/pc2/core/InternalController.java

```diff
@@ -3318,7 +3490,7 @@ public void sendServerLoginRequest(int inSiteNumber) throws Exception {
                 // remoteSite may have changed from inSiteNumber, check to see if we
                 // already know the connectionHandlerID for the server.
                 ClientId clientId = new ClientId(remoteSite.getSiteNumber(), ClientType.Type.SERVER, 0);
-                connectionHandlerID = contest.getConnectionHandleID(clientId);
+                connectionHandlerID = contest.getConnectionHandlerIDs(clientId).nextElement(); //ClientIds of type SERVER should have one, and only one, ConnectionHandlerID
```

Comment:
`contest.getConnectionHandlerIDs(ClientId)` returns the result of `LoginList.getConnectionHandlerIDs(ClientId)`, which never returns null -- as stated in its javadoc, it returns "a (possibly empty) Enumeration". 
But you did cause me to discover that the Javadoc for `IInternalContest` did not accurately reflect this -- it explicitly stated that it could return null.  I've pushed an update to that Javadoc.

I could also add a check that the returned value, from the method which is documented to never return null, is not null.  But I think I'll skip that for now.